src/exprs.c

```diff
@@ -919,9 +919,19 @@ Obj             EvalPermExpr (
 **
 **  'EvalListExpr'  just  calls 'ListExpr1'  and  'ListExpr2' to evaluate the
 **  list expression.
+**
+**  The 'safe' argument to ListExpr1 and ListExpr2 should be set to 1 when
+**  the list expression may be modified when its elements are being
+**  constructed. This can occur when an element of the list refers to ~.
+**
+**  When 'safe' is 1, we use a slower code path which ensures the list is
+**  in a valid state after each element is added, and we don't crash if the
+**  size or TNUM of the list is changed during the construction of elements
+**  in the list.
```

Comment:
Perhaps that should say "we don't crash if the size or type of the list changed, or even if gets swapped for a non-list object" ? After all, the user could use SWAP_OBJ to replace it by, say, a bigint.

OTOH, perhaps I am just overthinking that; and people using SWAP_OBJ deserve what they get ;-).